setup.py

```diff
@@ -268,6 +269,17 @@ def get_ethtool_macro():
     if SUNOS:
         posix_extension.libraries.append('socket')
         if platform.release() == '5.10':
+            # To ensure no regression, we assume by default
+            # the target is new_mib_compliant
+            new_mib_compliant = True
+            # See https://serverfault.com/q/524883
+            # for an explanation of Solaris /etc/release
+            with open('/etc/release') as f:
+                update = re.search(r'(?<=s10s_u)[0-9]{1,2}', f.readline())
+                if update is None or (int(update.group(0)) < 4):
+                    new_mib_compliant = False
+            if new_mib_compliant:
+                posix_extension.define_macros.append(('NEW_MIB_COMPLIANT', 1))
```

Comment:
How about shortening it a bit?

```python
            # Detect Solaris 5.10, update >= 4, see: 
            # https://github.com/giampaolo/psutil/pull/1638
            # For an explanation of Solaris /etc/release see:
            # https://serverfault.com/q/524883
            with open('/etc/release') as f:
                update = re.search(r'(?<=s10s_u)[0-9]{1,2}', f.readline())
                if update is None or (int(update.group(0)) >= 4):
                    posix_extension.define_macros.append(('NEW_MIB_COMPLIANT', 1))
```

Reply:
Right! But I still need to add the macro for the other platforms.

Question, is it on purpose that you've let the condition as `if update is None or …` (which now means that when no update is found or the update is greater than 4, the platform is new mib compliant)? Difference with my version is that I assumed when no update is found, the platform is not mib compliant.

Reply:
> Question, is it on purpose that you've let the condition as if update is None or … (which now means that when no update is found or the update is greater than 4, the platform is new mib compliant)?

No I just copy/pasted your code.